app/grants/views.py

```diff
@@ -615,3 +615,24 @@ def leaderboard(request):
             params['items'].append(item)
             counter += 1
     return TemplateResponse(request, 'grants/leaderboard.html', params)
+
+
+@csrf_exempt
+def new_matching_partner(request):
+
+    def get_json_response(message, status):
+        return JsonResponse(
+            {'status': status, 'message': message},
+            status=status
+        )
+
+    if not request.user.is_authenticated:
+        return get_json_response("Not Authorized", 403)
+
+    if request.POST:
+        MatchPledge(
+            profile=get_profile(request),
+            amount=request.POST.get('amount', 0)
```

Comment:
what if the user just fakes a post to this end point? 

IIRC the original spec requires some handling of the txid such that one could verify that the tx has cleared.  

at the very least we should store the `data` blob

Reply:
Could you tell me in which model i should store details about transaction?

Reply:
> what if the user just fakes a post to this end point?
> 
I've added basic transaction verification, but i'm not sure if it will be enough.

```
    if request.POST and tx_hash and profile:
        web3 = get_web3('mainnet')
        tx = web3.eth.getTransaction(tx_hash)
        if tx.to == gitcoin_account:
            .... save MatchPledge
```

Reply:
the MatchPledge model is fine.

i think if you verify the to address, the amount, and the tx has cleared, that should be sufficient

Reply:
thanks